### PR TITLE
Factor out the JWT verifier CLI options

### PR DESCRIPTION
**Pull request by aherrmann-da**
As suggested in https://github.com/digital-asset/daml/pull/7773#discussion_r509427760.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/cli-opts/src/main/scala/ledger-service/jwt/JwtVerifierConfigurationCli.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import java.nio.file.Paths
+
+import com.auth0.jwt.algorithms.Algorithm
+
+import scala.util.Try
+
+object JwtVerifierConfigurationCli {
+  def parse[C](parser: scopt.OptionParser[C])(setter: (JwtVerifierBase, C) => C): Unit = {
+    def setJwtVerifier(jwtVerifier: JwtVerifierBase, c: C): C = setter(jwtVerifier, c)
```
Comment (S11001001) [inline ledger-service/cli-opts/src/main/scala/ledger-service/jwt/JwtVerifierConfigurationCli.scala:14]:
```suggestion
    val setJwtVerifier = setter
```

**Review by S11001001 — APPROVED** [review]

**Review by cocreature — APPROVED** [review]
Nice, thank you!

File: ledger/sandbox-common/BUILD.bazel [hunk 2]
```diff
@@ -27,6 +27,7 @@ da_scala_library(
     deps = [
         "//daml-lf/data",
         "//language-support/scala/bindings",
+        "//ledger-service/cli-opts",
```
Comment (stefanobaghino-da) [inline ledger/sandbox-common/BUILD.bazel:30]: Probably `cli-opts` belongs in a more "neutral" directory. Having the sandbox depend on something under the JSON API sub-tree is valid but feels a bit weird.

**Review by stefanobaghino-da — APPROVED** [review]
I have one doubt about where we're placing the `cli-opts` package. If it's supposed to be used by several components it probably doesn't belong under the JSON API sub-tree.

File: ledger/sandbox-common/BUILD.bazel (hunk 2 above)
Comment (aherrmann-da) [reply ledger/sandbox-common/BUILD.bazel:30]: Happy to move this somewhere else. How about `//libs-scala/cli-opts`? That would also be an opportunity to split up http, tls, and jwt into dedicated Bazel targets. Though, in that case I'd prefer to do this in a follow-up PR.

Comment (stefanobaghino-da) [reply ledger/sandbox-common/BUILD.bazel:30]: I've always seen `//libs-scala` as more of a repository for libraries that are not necessarily related to the ledger. Maybe under `//ledger`?

Comment (stefanobaghino-da) [reply ledger/sandbox-common/BUILD.bazel:30]: @gerolf-da @cocreature Feel free to shut me down on this, maybe I'm being too nit-picking here. 😉

Comment (aherrmann-da) [reply ledger/sandbox-common/BUILD.bazel:30]: I'll merge this one as is as the increased test timeout will help with CI stability on master and will look into a better label in a follow up

Comment (stefanobaghino-da) [reply ledger/sandbox-common/BUILD.bazel:30]: 🙇🏻

**Pull request merged**